server.js

```diff
@@ -183,12 +184,17 @@ app.use([
             }
         }
     }),
+    (req, res, next) => {
```

Comment:
We don't need this extra middleware, it can go into the preview middleware assuming we call the CSP stuff first.

```js
module.exports = (req, res, next) => {
    const isContentPreview =
        req.query.token &&
        (req.query['x-craft-live-preview'] || req.query['x-craft-preview']);

    if (isContentPreview) {
        res.locals.isContentPreview = true;
        res.cacheControl = { noStore: true };
        res.removeHeader('X-Frame-Options');
        requireStaffAuth(req, res, next);
    } else {
        next();
    }
}
```

Reply:
Thinking about it we might even want to use CSPs `frame-ancestors` in favour of this https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors as `X-Frame-Options` is largely deprecated.

Reply:
Might leave it as-is for now as we get into slightly fiddly config territory where technically the live website might be loaded into the test CMS for previews but its iframe header won't allow it (unless we configured the header to allow all CMS domains for all app environments).